common/src/main/java/us/myles/ViaVersion/protocols/protocol1_13to1_12_2/ChatRewriter.java

```diff
@@ -1,96 +1,43 @@
 
-    // Based on https://github.com/SpigotMC/BungeeCord/blob/master/chat/src/main/java/net/md_5/bungee/api/chat/TextComponent.java
-    public static String fromLegacyTextAsString(String message, ChatColor defaultColor, boolean itemData) {
-        TextComponent headComponent = new TextComponent();
-        TextComponent component = new TextComponent();
-        StringBuilder builder = new StringBuilder();
-        if (itemData) {
-            // Workaround for all italic lore
-            headComponent.setItalic(false);
-            //TODO set first child to italics if it doesn't have a color
-        }
-
-        for (int i = 0; i < message.length(); i++) {
-            char c = message.charAt(i);
-            if (c == ChatColor.COLOR_CHAR) {
-                if (++i >= message.length()) {
-                    break;
-                }
-                c = message.charAt(i);
-                if (c >= 'A' && c <= 'Z') {
-                    c += 32;
-                }
-                ChatColor format = ChatColor.getByChar(c);
-                if (format == null) {
-                    continue;
-                }
-                if (builder.length() > 0) {
-                    TextComponent old = component;
-                    component = new TextComponent(old);
-                    old.setText(builder.toString());
-                    builder = new StringBuilder();
-                    headComponent.addExtra(old);
-                }
-                if (ChatColor.BOLD.equals(format)) {
-                    component.setBold(true);
-                } else if (ChatColor.ITALIC.equals(format)) {
-                    component.setItalic(true);
-                } else if (ChatColor.UNDERLINE.equals(format)) {
-                    component.setUnderlined(true);
-                } else if (ChatColor.STRIKETHROUGH.equals(format)) {
-                    component.setStrikethrough(true);
-                } else if (ChatColor.MAGIC.equals(format)) {
-                    component.setObfuscated(true);
-                } else if (ChatColor.RESET.equals(format)) {
-                    format = defaultColor;
-
-                    component = new TextComponent();
-                    component.setColor(format);
-                } else {
-                    component = new TextComponent();
-                    component.setColor(format);
-                }
-                continue;
+    public static String legacyTextToJsonString(String message, boolean itemData) {
+        Component component = Component.text(builder -> {
+            if (itemData) {
+                builder.decoration(TextDecoration.ITALIC, false);
             }
-            builder.append(c);
-        }
-
-        component.setText(builder.toString());
-        headComponent.addExtra(component);
-
-        return ComponentSerializer.toString(headComponent);
+            builder.append(LegacyComponentSerializer.legacySection().deserialize(message));
```

Comment:
How are links handled?

Reply:
Doesn't matter, the method is not used for chat messages.